### PR TITLE
Add MLTransform IT test

### PR DESCRIPTION
**Pull request by AnandInguva**
part of  https://github.com/apache/beam/issues/27626

This only add IT test. I removed performance tests from this PR since they are OOMing which shouldn't happen. I will work on them when I come back of OOO.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by jrmccluskey** [comment]
Run Python PreCommit

**Comment by damccorm** [comment]
Pulling in the latest from master should fix the workflow failures FWIW

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

File: sdks/python/apache_beam/examples/ml_transform/vocab_tfidf_processing.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        pcoll
+        | "ReadPositiveExample" >> beam.io.ReadFromText(self.pos_file_pattern)
+        | 'PairWithOne' >> beam.Map(lambda review: (review, 1)))
+
+    all_examples = ((negative_examples, positive_examples)
+                    | 'FlattenPColls' >> beam.Flatten())
+
+    shuffled_examples = (
+        all_examples
+        | 'Distinct' >> beam.Distinct()
```
[the remark this answers is not in the document]
Comment (AnandInguva) [reply sdks/python/apache_beam/examples/ml_transform/vocab_tfidf_processing.py]: Moved it

File: sdks/python/apache_beam/examples/ml_transform/ml_transform_it_test.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    extra_opts['job_name'] = 'mltransform-large-movie-review-dataset-{}'.format(
+        uuid.uuid4().hex)
+
+    test_pipeline = TestPipeline(is_integration_test=True)
+    start_time = time.time()
+    vocab_tfidf_processing.run(
+        test_pipeline.get_full_options_as_args(
+            **extra_opts, save_main_session=False),
+    )
+    end_time = time.time()
```
[the remark this answers is not in the document]
Comment (AnandInguva) [reply sdks/python/apache_beam/examples/ml_transform/ml_transform_it_test.py:80]: Added validation on the checking the generated vocab

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

File: sdks/python/apache_beam/examples/ml_transform/vocab_tfidf_processing.py [hunk 3]
```diff
@@ -130,7 +130,7 @@ def preprocess_data(
   data_pcoll = (
       data_pcoll | beam.ParDo(MapTFIDFScoreToVocab(artifact_location)))
 
-  _ = (data_pcoll | beam.io.WriteToText(output_dir))
+  #   _ = (data_pcoll | beam.io.WriteToText(output_dir))
```
Comment (damccorm) [inline sdks/python/apache_beam/examples/ml_transform/vocab_tfidf_processing.py:133]: Can we get rid of this comment?

Comment (damccorm) [reply sdks/python/apache_beam/examples/ml_transform/vocab_tfidf_processing.py:133]: Actually, is there a reason we're not writing anymore?

File: sdks/python/apache_beam/testing/benchmarks/cloudml/cloudml_benchmark_test.py [hunk 4]
```diff
@@ -55,7 +55,7 @@ def _publish_metrics(pipeline, metric_value, metrics_table, metric_name):
   )])
 
 
-@pytest.mark.uses_tft
+# @pytest.mark.uses_tft
```
Comment (damccorm) [inline sdks/python/apache_beam/testing/benchmarks/cloudml/cloudml_benchmark_test.py:58]: Can we just get rid of this comment (and the pytest import comment)

Comment (damccorm) [reply sdks/python/apache_beam/testing/benchmarks/cloudml/cloudml_benchmark_test.py:58]: Wait, don't we actually need this for https://github.com/apache/beam/blob/6cae195d4ab62a5b433f8781328ddc3f57b9c74f/sdks/python/test-suites/dataflow/common.gradle#L454

Comment (jrmccluskey) [reply sdks/python/apache_beam/testing/benchmarks/cloudml/cloudml_benchmark_test.py:58]: Yes that is needed for testing, it won't be picked up in that suite if absent or commented out

Comment (AnandInguva) [reply sdks/python/apache_beam/testing/benchmarks/cloudml/cloudml_benchmark_test.py:58]: ah no. I will revert it once the expected test runs on Jenkins and is green

File: sdks/python/apache_beam/examples/ml_transform/vocab_tfidf_processing.py (hunk 3 above)
Comment (AnandInguva) [reply sdks/python/apache_beam/examples/ml_transform/vocab_tfidf_processing.py:133]: This should be uncommented. I forgot to remove it/

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

**Comment by AnandInguva** [comment]
Run TFT Criteo Benchmarks

File: sdks/python/apache_beam/testing/benchmarks/cloudml/cloudml_benchmark_test.py (hunk 4 above)
Comment (AnandInguva) [reply sdks/python/apache_beam/testing/benchmarks/cloudml/cloudml_benchmark_test.py]: Reverted the change.

**Comment by AnandInguva** [comment]
Passing test - https://ci-beam.apache.org/job/beam_CloudML_Benchmarks_Dataflow_PR/46/

**Review by damccorm — APPROVED** [review]
LGTM, thanks

**Comment by AnandInguva** [comment]
Failed tests are not relevant. Merging.

**Pull request merged**